### ISSUE TITLE
doctest failed on vectorize...

### ISSUE THREAD

**Issue by gershwinlover** [iit-cs579/assignments#24]
When I explicitly define the CSR matrix with "dtype=np.int64" the output looks like:
Expected:
    array([[1, 0, 1, 1, 1, 1],
           [0, 2, 0, 1, 0, 0]], dtype=int64)
Got:
    array([[1, 0, 1, 1, 1, 1],
           [0, 2, 0, 1, 0, 0]])
However, when I define it as "dtype=np.int32", the output looks like:
Expected:
    array([[1, 0, 1, 1, 1, 1],
           [0, 2, 0, 1, 0, 0]], dtype=int64)
Got:
    array([[1, 0, 1, 1, 1, 1],
           [0, 2, 0, 1, 0, 0]], dtype=int32)
Anyone has idea how I can pass this doctest?

**Comment by vchariharan** [comment]
Hi, 
I guess you have not properly used it.. you maybe overriding csr matrix.. you get style=int32 cos it comes as default for given csr matrix.. try removing it and you ll see the same.. 

I have used style=no.int64 and have got output as expected and doctest passing.

You pls cross check csr_matrix assignment and see if you are assigning only once

**Comment by vchariharan** [comment]
It's dtype and not style in my comment above.. damn autocorrect

**Comment by gershwinlover** [comment]
Thanks for your kind help! However, I have tried set it dtype = np.int64, it gives me:
Expected:
array([[1, 0, 1, 1, 1, 1],
[0, 2, 0, 1, 0, 0]], dtype=int64)
Got:
array([[1, 0, 1, 1, 1, 1],
[0, 2, 0, 1, 0, 0]])
which is the same as doing dtype =  int, or leaving it empty (using the default)

**Comment by gershwinlover** [comment]
anyone has an idea?

**Comment by gershwinlover** [comment]
Okay, after a whole day long debugging, I finally found out how to get ride of this problem (but still don't know the reason).  

There are 3 examples in the given link showing 3 different ways of creating a csr_matrix https://goo.gl/f5TiF1. I followed the last one.  However, this particular way doesn't show dtype if it's the default dtype; whereas the second way, even if you don't explicitly state dtype=np.int64, which is the default, it will still show dtype=int64!

**However**, I look into this question not only because the failed doctest, but that my output is different than Log.txt. Thanks to Hanqiao's help. We finally found out that it's not because of my csr_matrix. It's because of the KFold!

The description in a2.py doesn't say what k we should use for KFold in eval_all_combinations. So I used 10. But this k must be 5 in order to get the same result as Log.txt!!! 

**Just want to post these for your reference. Hopefully these words can help saving some time on debugging**
